chips/stm32f4xx/src/gpio.rs

```diff
@@ -865,6 +866,25 @@ impl<'a> Pin<'a> {
         self.pinid
     }
 
+    pub unsafe fn enable_interrupt(&'static self) {
+        let interrupt = match self.pinid.get_pin_number() {
+            0 => nvic::EXTI0,
+            1 => nvic::EXTI1,
+            2 => nvic::EXTI2,
+            3 => nvic::EXTI3,
+            4 => nvic::EXTI4,
+            5..=9 => nvic::EXTI9_5,
+            10..=15 => nvic::EXTI15_10,
+            _ => panic!("Pin {} is not valid", self.pinid.get_pin_number()),
+        };
+
+        let exti_line_id = LineId::from_u8(self.pinid.get_pin_number() as u8).unwrap();
+
+        self.exti.associate_line_gpiopin(exti_line_id, &self);
+        // enable interrupt
+        cortexm4::nvic::Nvic::new(interrupt).enable();
+    }
+
```

Comment:
cc @bradjc This is similar to the imx PR so wanted to call your attention to it. I think I recall reviewing a PR (I think one from @alexandruradovici ) awhile back where always enabling all interrupts at the nvic level was problematic, but I cannot remember if it was for these boards or something else.

Reply:
That the for the Raspberry Pi Pico :)

For the stm, this is how the original author did it. @valexandru took the stm as an inspiration for the imx. I'll try enabling all the interrupts and see if it works.

Reply:
I enabled all the interrupts in init and it seems to work.

Reply:
Great, I think that is a much better design.